src/app/utils/api.class.ts

```diff
@@ -0,0 +1,33 @@
+import axios, { type AxiosInstance, type AxiosRequestConfig } from 'axios';
+
+const instance = axios.create({
+  baseURL: 'http://localhost:3000/api/v1',
+});
+
+interface IApiProps {
+  url: string;
+  data: object;
+  params?: AxiosRequestConfig<any>;
+}
+
+class Api {
```

Comment:
Perhaps we can make this as a base class and make service specifics classes for shlinks, shlink-endpoints, etc so that we do not need to pass the url in the components. This will also help with refactoring should we look to use different endpoint names in future

Reply:
It's a good idea :rocket: